### PR TITLE
Merge #548 and #552

### PR DESCRIPTION
**Pull request by jas32096**
Closes #548 and #552

### CODE REVIEW

**Comment by jas32096** [comment]
Alright. Should be working now, with the new syntax.

Note: For invoking from the command line, you'll need to surround the search with quotes like:
&nbsp;&nbsp;&nbsp;&nbsp;`mpsyt '/best music -d long -a year'`

**Comment by ids1024** [comment]
I found an issue with this. If you try a command like `/query --some-invalid-argument` or `/query --help`, it prints a message then exits. The message is fine, but it shouldn't terminate mpsyt.

I don't know if there is a way to change that behavior with `argparse`. I don't see anything in the docs.

**Comment by jas32096** [comment]
Right. I also forgot about passing on extra args with, kwargs, but I'm not gunna do that, for the time being. I'll try surrounding the parse with a try/except

**Comment by jas32096** [comment]
Made some changes and really tried to beat it up this time.

One of the last things I tested with was `/ --this -qwerty -a 2015-09-22 this is a test --duration long --bad -c --`, which is was able to handle fine, ignoring the unused arguments.

File: mps_youtube/commands/search.py [hunk 1]
```diff
@@ -218,10 +218,14 @@ def related_search(vitem):
...
-    args = parser.parse_args(term.split(' '))
-    videoDuration = args.duration if args.duration else 'any'
-    after = args.after
-    term = ' '.join(args.search)
+    try:     #TODO make use of unkowns
+        args, unknown = parser.parse_known_args(term.split(' '))
+        videoDuration = args.duration if args.duration else 'any'
+        after = args.after
+        term = ' '.join(args.search)
+    except SystemExit as e:  #<------ argsparse calls exit()
```
Comment (ids1024) [inline mps_youtube/commands/search.py]: Oh, I didn't know that `sys.exit()` triggers a catchable exception. Nice!

**Pull request merged**